### ISSUE TITLE
Snap: Failed to read user shell environment

### ISSUE THREAD

**Issue by NatoBoram** [microsoft/vscode#72042]
- VSCode Version: `1.34.0-insider` 56f1b4795fed05677eb82a21dac4fc1ab62d747b x64
- OS Version: `Ubuntu 18.04.2 LTS x86_64`

Steps to Reproduce:

1. Install VSCode as a snap
2. Install JSHint
3. Install `dbaeumer.jshint`

Does this issue occur when all extensions are disabled?: N/A

I also reported it in https://github.com/Microsoft/vscode-jshint/issues/71, but it's probably unrelated to the extension itself.

**Comment by joaomoreno** [comment]
@dbaeumer What can the user do to figure out what's wrong?

**Comment by dbaeumer** [comment]
@RMacfarlane took that over. We might want to consider changing the publisher

**Comment by dbaeumer** [comment]
If there anything in the JSHint output panel?

**Comment by NatoBoram** [comment]
The **Output** tab is empty. All I got is the error message.

```markdown
Failed to load jshint library. Please install jshint in your workspace folder using `npm install jshint` or globally using `npm install -g jshint` and then press Retry.
```

**Comment by dbaeumer** [comment]
@NatoBoram do you have jshint itself installed? Your workspace doesn't have a node_modules folder.

**Comment by joaomoreno** [comment]
Also, can you reproduce with VS Code without Snap? https://update.code.visualstudio.com/latest/linux-x64/insider

**Comment by NatoBoram** [comment]
@dbaeumer JSHint was installed globally.
```bash
jshint --version
jshint v2.10.2
```

@joaomoreno
No. The `.deb` works fine.

**Comment by dbaeumer** [comment]
@NatoBoram does it work if you install jshint locally?

**Comment by NatoBoram** [comment]
Yes, it works when installed locally.

**Comment by joaomoreno** [comment]
@NatoBoram Do you use NVM? If you open the Developer Tools and run `getLazyEnv().then(console.log)`, do you get the expected environment?

**Comment by dbaeumer** [comment]
Reopening.

@joaomoreno  and I looked into this and the problem seems to be that the environment is not correctly sourced when running out of a snap which makes the server fail looking up global node moddules.

**Comment by alireza4050** [comment]
Also faced this issue. the problem is with gnome launcher .desktop file. it does not run ~/.bashrc or any other shell environment file. running 'code' from terminal inherits the environment and works just fine.

**Comment by sgrf** [comment]
I have a similar issue but the above does not fix it. 
If I add VSCode as a favorite from the UBUNTU dash and then click VSCode icon on the sidebar none of my scripts work in VSCode script explorer. I get npm not found.
If I start VSCode within a terminal everything works.

**Comment by sgrf** [comment]
I should have said "none of my npm scripts work

**Comment by joaomoreno** [comment]
Root cause: https://forum.snapcraft.io/t/new-store-automated-review-rule-is-incorrect/12963/27?u=joaomoreno

Fix is coming in from the snapcraft team

**Comment by joaomoreno** [comment]
Fixed by #80585

cc @alexr00
